src/components/modal/modal.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import closeIcon from "../../assets/closeIcon.svg";
+import { motion, AnimatePresence, useMotionValue } from "framer-motion";
+import useMediaQuery from "../../utils/useMediaQuery";
+/**
+ The **Modal** component is the base of all the modals in the web app.
+ *
+ *  @author [Nuriya](https://github.com/NuriyaAkh)
+ */
+const ModalComponent = styled(motion.div)`
+  box-sizing: border-box;
+  justify-content: center;
+  align-items: center;
```

Comment:
Use Bootstrap classes instead of style declarations where possible (position, alignment, flex display, etc) for all components